pkg/proxy/iptables/proxier_test.go

```diff
@@ -1345,6 +1347,270 @@ func assertIPTablesRulesNotEqual(t *testing.T, expected, result string) {
 	}
 }
 
+// cidrContains checks if cidrStr contains ipStr. cidrStr can actually be a CIDR or an IP
+func cidrContains(cidrStr, ipStr string) bool {
+	ip := netutils.ParseIPSloppy(ipStr)
+	if ip == nil {
+		panic("bad IP in test case: " + ipStr)
+	}
+
+	if strings.Contains(cidrStr, "/") {
+		_, cidr, err := netutils.ParseCIDRSloppy(cidrStr)
+		if err != nil {
+			panic("bad CIDR in kube-proxy output: " + err.Error())
+		}
+		return cidr.Contains(ip)
+	} else {
+		ip2 := netutils.ParseIPSloppy(cidrStr)
+		if ip2 == nil {
+			panic("bad IP in kube-proxy output: " + cidrStr)
+		}
+		return ip.Equal(ip2)
+	}
+}
+
+var addChainRegex = regexp.MustCompile(`-A ([^ ]*) `)
+var commentRegex = regexp.MustCompile(`-m comment --comment ("[^"]*"|[^" ]*) `)
+var moduleRegex = regexp.MustCompile("-m ([^ ]*)")
+var localRegex = regexp.MustCompile("(!)? --(src|dst)-type LOCAL")
+var destIPRegex = regexp.MustCompile("(!)? -d ([^ ]*) ")
+var destPortRegex = regexp.MustCompile(" --dport ([^ ]*) ")
+var sourceIPRegex = regexp.MustCompile("(!)? -s ([^ ]*) ")
+var affinityRegex = regexp.MustCompile(" --rcheck ")
+var probabilityRegex = regexp.MustCompile(" --probability ([^ ]*) ")
+var jumpRegex = regexp.MustCompile(" -j ([^ ]*)( --to-destination (.*))?$")
+
+var noMatchModules = sets.NewString("conntrack", "mark")
+var knownModules = sets.NewString("addrtype", "recent", "statistic", "tcp")
+
+type iptablesTracer struct {
+	// tables is a map of table name to table. Each table is a map from chain name
+	// to the rules in that chain
+	tables map[string]map[string][]string
```

Comment:
can we use some types

type rules []string
type chains map[string]rules 

tables map[string]chains